### PR TITLE
Panels refactoring

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Split the `panel.js` file, rename some of the components, prune unused ones, share more code and enable eslint and flow on them.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal changes

- [X] **DONE**

## Test coverage
- No tests: the existing tests should be validating these changes

- [X] **DONE**

## Links

Note that this PR is based on PR #216 and thus includes its changes.

- [X] **DONE**

### CODE REVIEW

**Comment by cbosdo** [comment]
gitarro rerun changelog_test !!!

File: web/spacewalk-web.changes [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- Refactor the panel.js components
```
Comment (lneves12) [inline web/spacewalk-web.changes]: hum I would not add anything to the changes here. Last retrospective we discussed that only bugs or changes that are visible to the costumer should be presented here

Comment (ncounter) [reply web/spacewalk-web.changes]: Yeah Maintenance team approached us asking to put more value to our changelog entries, in terms of "the changelog should be understandable by the customer, and should be deliver value to him". This PR instead, is about internal refactoring only.

Comment (lneves12) [reply web/spacewalk-web.changes]: nope it just a refactor, I would remove it

**Comment by cbosdo** [comment]
gitarro no changelog needed !!!

**Pull request merged**